src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -82,6 +87,23 @@ static CurlHandler()
             // Feature detection
             s_supportsSSL = (CurlFeatures.CURL_VERSION_SSL & s_curlVersionInfoData.features) != 0;
             s_supportsAutomaticDecompression = (CurlFeatures.CURL_VERSION_LIBZ & s_curlVersionInfoData.features) != 0;
+
+            // We can send any pointer, even null seems to work. But sending a non-null value, just in case
+            if (CurlSslBackend.AllowsSslCallback(s_curlVersionInfoData.ssl_version))
+            {
+                s_clientCertificatePolicy = ClientCertificatePolicy.Callback;
+                VerboseTrace("openSSL compatible");
+            }
+            else if (CurlSslBackend.AllowsClientCertCommonName(s_curlVersionInfoData.ssl_version))
```

Comment:
If we can't make the behavior of the code Obvious and Predictable, it feels to me like it should throw PlatformNotSupportedException until such a time as we can work out the problems.

So, detecting which policy is in play here is fine; but then when you need to go utilize them it might be best to throw instead.

And if I understand correctly, that eliminates the need to export to PFX, since only libcurl+openssl would be supported?